### ISSUE TITLE
Twitter Provider doesn't work

### ISSUE THREAD

**Issue by brezhitskiy42** [strapi/strapi#13217]
### Required System information

- Node.js version: 16.14.2
- NPM version: 8.5.0
- Strapi version: 4.1.9
- Database: PostgreSQL 14
- Operating system: Ubuntu 20.04 (LTS) x64

### Bug description

Twitter provider doesn't work. When requesting `/api/auth/twitter/callback?access_token=ACCESS_TOKEN&access_secret=ACCESS_SECRET`, Strapi return `400 Error`.

The error seems to occur here - https://github.com/strapi/strapi/blob/master/packages/plugins/users-permissions/server/services/providers-list.js#L118, because this part return `{"errors":[{"message":"Bad Authentication data","code":215}]}`.

![image](https://user-images.githubusercontent.com/10079097/166112669-495e2231-be95-4a8b-9325-a6c4a2344829.png)

Perhaps the problem is that the `Client ID` and the `Client Secret` are not passed.

![image](https://user-images.githubusercontent.com/10079097/166112919-42b9e3c1-28a9-4bdf-9e23-e1f29798a933.png)

Because if we explicitly pass `oauth` options to the `purest` function, everything works. For example:
```js
const twitter = purest({
  provider: 'twitter',
  defaults: {
    oauth: {
      consumer_key: 'Client ID',
      consumer_secret: 'Client Secret'
    }
  }
});
```

![image](https://user-images.githubusercontent.com/10079097/166113252-fae4e6d6-9752-4df6-a230-8b8881077697.png)

**Comment by kwang1198** [comment]
@brezhitskiy42 have you tried to pass your client id/secret in `.auth('ID', 'SECRET')`? Or print out `query.access_secret` to make sure if the value is correct.

**Comment by brezhitskiy42** [comment]
@kwang1198 The value of `query.access_secret` is correct.

Shouldn't these values be passed automatically? (`Client ID` and `Client Secret`)

**Comment by kwang1198** [comment]
@brezhitskiy42 did you get the `access_token` and `access_secret` from Twitter oauth callback? These values should be passed as parameters from Twitter Oauth Callback.

**Comment by brezhitskiy42** [comment]
@kwang1198 Yes, I wrote about it above - When requesting `/api/auth/twitter/callback?access_token=ACCESS_TOKEN&access_secret=ACCESS_SECRET`, Strapi return `400 Error`.

**Comment by kwang1198** [comment]
After checking previous versions using `purest@3.x.x`, I think consumer key and secret is needed for accessing the Twitter API as pointed out in [here](https://github.com/simov/purest/issues/24).
Thus the code should be:
```javascript
const twitter = purest({
  provider: 'twitter',
  defaults: {
    oauth: {
      consumer_key: providers.twitter.key,
      consumer_secret:  providers.twitter.secret
    }
  }
});
```

**Comment by GraemeFulton** [comment]
I can't upgrade past 4.1.7 because this issue - I've seen it is fixed here: https://github.com/strapi/strapi/commit/70b05b5e9b64a97b9b00761783b5536b750d8235

There is the same issue with GitHub... when will this be merged into the main version? I don't know how to override it, so am stuck on 4.1.7
